src/overlays/actors/ovl_En_Kgy/z_en_kgy.c

```diff
@@ -41,105 +63,1132 @@ const ActorInit En_Kgy_InitVars = {
+}
+
+void func_80B40B38(EnKgy* this, s16 arg1, u8 arg2, f32 arg3) {
+    static AnimationHeader* D_80B43270[] = {
+        &D_06004B98, &D_060008FC, &D_0600292C, &D_060042E4, &D_060101F0,
+        &D_06001764, &D_06003334, &D_06010B84, &D_06001EA4, &D_06003D88,
+    };
+
+    SkelAnime_ChangeAnim(&this->skelAnime, D_80B43270[arg1], 1.0f, 0.0f,
+                         SkelAnime_GetFrameCount(&D_80B43270[arg1]->common), arg2, arg3);
+    this->unk_2D2 = arg1;
+}
+
+void func_80B40BC0(EnKgy* this, s16 arg1) {
+    if (arg1 != this->unk_2D2) {
+        func_80B40B38(this, arg1, 0, -5.0f);
+    }
+}
+
+EnKbt* func_80B40BFC(GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_NPC].first;
+
+    while (actor != NULL) {
+        if (actor->id == ACTOR_EN_KBT) {
+            return (EnKbt*)actor;
+        }
+        actor = actor->next;
+    }
+    return NULL;
+}
+
+ObjIcePoly* func_80B40C38(GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_ITEMACTION].first;
+
+    while (actor != NULL) {
+        if (actor->id == ACTOR_OBJ_ICE_POLY) {
+            return (ObjIcePoly*)actor;
+        }
+        actor = actor->next;
+    }
+    return NULL;
+}
+
+void func_80B40C74(GlobalContext* globalCtx) {
+    gSaveContext.roomInf[globalCtx->sceneNum][5] |= 1;
+    if ((gSaveContext.day % 5) == 1) {
+        gSaveContext.roomInf[globalCtx->sceneNum][5] |= 2;
+    } else {
+        gSaveContext.roomInf[globalCtx->sceneNum][5] &= ~2;
+    }
+}
+
+void func_80B40D00(GlobalContext* globalCtx) {
+    gSaveContext.roomInf[globalCtx->sceneNum][5] |= 4;
+}
+
+void func_80B40D30(GlobalContext* globalCtx) {
+    gSaveContext.roomInf[globalCtx->sceneNum][5] &= ~7;
+}
+
+s32 func_80B40D64(GlobalContext* globalCtx) {
+    return gSaveContext.roomInf[globalCtx->sceneNum][5] & 1;
+}
+
+s32 func_80B40D8C(GlobalContext* globalCtx) {
+    return gSaveContext.roomInf[globalCtx->sceneNum][5] & 4;
+}
+
+s32 func_80B40DB4(GlobalContext* globalCtx) {
+    if (((gSaveContext.day % 5) == 3) ||
+        (((gSaveContext.day % 5) == 2) && (gSaveContext.roomInf[globalCtx->sceneNum][5] & 2))) {
+        return true;
+    }
+    return false;
+}
+
+void func_80B40E18(EnKgy* this, u16 arg1) {
+    if (this->unk_2A0 != NULL) {
+        this->unk_2A0->actor.textId = arg1;
+    }
+}
+
+void func_80B40E38(EnKgy* this) {
+    if (this->unk_2A0 != NULL) {
+        this->unk_2A0->actor.home.rot.z = 1;
+    }
+}
+
+s32 func_80B40E54(EnKgy* this) {
+    if (this->unk_2A0 != NULL) {
+        return this->unk_2A0->actor.home.rot.z;
+    }
+    return 0;
+}
+
+void func_80B40E74(EnKgy* this, GlobalContext* globalCtx, u16 arg2) {
+    func_80151938(globalCtx, arg2);
+    this->actor.textId = arg2;
+    func_80B40E18(this, this->actor.textId);
+}
+
+void func_80B40EBC(EnKgy* this, GlobalContext* globalCtx, u16 arg2) {
+    func_80B40E74(this, globalCtx, arg2 += 1);
```

Comment:
Yuck! does `++` work? I assume it can't go outside the func?

Reply:
Post-increment ++ doesn't work, it has to be updated before the function call. It's either `++arg2` or `arg += 1` and I thought the latter looks better. I can't move it before the function call, no.